polaris-react/src/components/ResourceItem/ResourceItem.tsx

```diff
@@ -217,70 +234,82 @@ class BaseResourceItem extends Component<CombinedProps, State> {
             </ButtonGroup>
           </div>
-        );
+        ) : null;
 
         const disclosureAccessibilityLabel = name
           ? i18n.translate('Polaris.ResourceList.Item.actionsDropdownLabel', {
               accessibilityLabel: name,
             })
           : i18n.translate('Polaris.ResourceList.Item.actionsDropdown');
 
-        disclosureMarkup = (
-          <div className={styles.Disclosure} onClick={stopPropagation}>
-            <Popover
-              activator={
-                <Button
-                  accessibilityLabel={disclosureAccessibilityLabel}
-                  onClick={this.handleActionsClick}
-                  plain
-                  icon={HorizontalDotsMinor}
-                />
-              }
-              onClose={this.handleCloseRequest}
-              active={actionsMenuVisible}
-            >
-              <ActionList items={shortcutActions} />
-            </Popover>
-          </div>
-        );
-      } else {
+        disclosureMarkup =
+          !selectMode && breakpoints?.lgDown ? (
+            <div onClick={stopPropagation}>
+              <Popover
+                activator={
+                  <Button
+                    accessibilityLabel={disclosureAccessibilityLabel}
+                    onClick={this.handleActionsClick}
+                    plain
+                    icon={HorizontalDotsMinor}
+                  />
+                }
+                onClose={this.handleCloseRequest}
+                active={actionsMenuVisible}
+              >
+                <ActionList items={shortcutActions} />
+              </Popover>
+            </div>
+          ) : null;
+      } else if (breakpoints?.lgUp) {
         actionsMarkup = (
           <div className={styles.Actions} onClick={stopPropagation}>
-            <ButtonGroup segmented>
-              {buttonsFrom(shortcutActions, {
-                size: 'slim',
-              })}
-            </ButtonGroup>
+            <Box position="absolute" insetBlockStart="4" insetInlineEnd="4">
+              <ButtonGroup segmented>
+                {buttonsFrom(shortcutActions, {size: 'slim'})}
+              </ButtonGroup>
+            </Box>
           </div>
         );
       }
     }
 
-    const content = children ? (
-      <div className={styles.Content}>{children}</div>
-    ) : null;
-
-    const containerClassName = classNames(
-      styles.Container,
-      verticalAlignment &&
-        styles[variationName('alignment', verticalAlignment)],
-    );
-
     const containerMarkup = (
-      <div className={containerClassName} id={this.props.id}>
-        {ownedMarkup}
-        {content}
-        {actionsMarkup}
-        {disclosureMarkup}
-      </div>
+      <Box
+        id={this.props.id}
+        position="relative"
+        padding="3"
+        paddingInlineStart={{xs: '4', sm: '5'}}
+        paddingInlineEnd={{xs: '4', sm: '5'}}
+        zIndex="var(--pc-resource-item-content-stacking-order)"
```

Comment:
No, going from 100 to 400 to 510 (z-1, z-2, z-3) just doesn't make sense for a localized stacking order. 

If there's a component that makes use of lots of z-indexes (these components exist) then we get up to pretty high numbers, pretty quickly. This will cause issues where elements start overlapping backdrops, popovers, navigation etc...